src/Microsoft.Data.Analysis/IDataView.Extension.cs

```diff
@@ -0,0 +1,126 @@
+                {
+                    columns.Add(new BooleanDataFrameColumn(column.Name));
+                }
+                else if (type == NumberDataViewType.Byte)
+                {
+                    columns.Add(new ByteDataFrameColumn(column.Name));
+                }
+                else if (type == NumberDataViewType.Double)
+                {
+                    columns.Add(new DoubleDataFrameColumn(column.Name));
+                }
+                else if (type == NumberDataViewType.Single)
+                {
+                    columns.Add(new SingleDataFrameColumn(column.Name));
+                }
+                else if (type == NumberDataViewType.Int32)
+                {
+                    columns.Add(new Int32DataFrameColumn(column.Name));
+                }
+                else if (type == NumberDataViewType.Int64)
+                {
+                    columns.Add(new Int64DataFrameColumn(column.Name));
+                }
+                else if (type == NumberDataViewType.SByte)
+                {
+                    columns.Add(new SByteDataFrameColumn(column.Name));
+                }
+                else if (type == NumberDataViewType.Int16)
+                {
+                    columns.Add(new Int16DataFrameColumn(column.Name));
+                }
+                else if (type == NumberDataViewType.UInt32)
+                {
+                    columns.Add(new UInt32DataFrameColumn(column.Name));
+                }
+                else if (type == NumberDataViewType.UInt64)
+                {
+                    columns.Add(new UInt64DataFrameColumn(column.Name));
+                }
+                else if (type == NumberDataViewType.UInt16)
+                {
+                    columns.Add(new UInt16DataFrameColumn(column.Name));
+                }
+                else if (type == TextDataViewType.Instance)
+                {
+                    columns.Add(new StringDataFrameColumn(column.Name));
+                }
+                else
+                {
+                    throw new NotSupportedException(String.Format(Microsoft.Data.Strings.NotSupportedColumnType, type.RawType.Name));
+                }
+            }
+
+            using (DataViewRowCursor cursor = dataView.GetRowCursor(activeColumns))
+            {
+                Delegate[] activeColumnDelegates = new Delegate[activeColumns.Count];
+                int columnIndex = 0;
+                foreach (DataViewSchema.Column column in activeColumns)
+                {
+                    Delegate valueGetter = columns[columnIndex].GetValueGetterUsingCursor(cursor, column);
+                    activeColumnDelegates[columnIndex] = valueGetter;
+                    columnIndex++;
+                }
+                while (cursor.MoveNext() && cursor.Position < maxRows)
+                {
+                    columnIndex = 0;
+                    foreach (DataViewSchema.Column column in activeColumns)
```

Comment:
If `column` isn't used in `AddValueUsingCursor`, this could be changed to a simple `for (int i = 0; i < activeColumnDelegates.Length; i++)`